### PR TITLE
`Encrypted Mnemonic Wallet`

### PR DESCRIPTION
**Pull request by ivan-zynesis**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Create pin after verify mnemonic words
OceanInterface prompt pin interface

#### Which issue(s) does this PR fixes?:

Fixes #

#### Additional comments?:
Considerations of current implementation
1. Passcode prompt page UI accessible globally
2. OceanInterface remained for async task status display (after wallet unlocked) ie slow internet, broadcast take very long or fail.
3. OceanInterface remained as child of AppNavigator, accessible by tabbed pages for submit transaction (unlike prompt UI)
4. Thus all interactions with prompt UI actually done via WalletManagementContext

### CODE REVIEW

File: app/api/scrypt.native.temp.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import rnScrypt from 'react-native-scrypt'
```
Comment (ivan-zynesis) [inline app/api/scrypt.native.temp.ts:1]: when I name this file as scrypt.native.ts

default platform detected as ios even running on browser for development. Need to figure out how to fix it

File: app/api/scrypt.native.temp.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  N: 16384,
+  r: 8,
+  p: 8
+}
+
+class NativeScryptModule implements ScryptProvider {
+  passphraseToKey (nfcUtf8: string, salt: Buffer, desiredKeyLen: number): Buffer {
+    let result: Buffer | null = null
+    let err: Error | null = null
+    rnScrypt(
```
Comment (ivan-zynesis) [inline app/api/scrypt.native.temp.ts:14]: Fit this into jellyfish expected platform agnostic interface

File: app/components/OceanInterface/OceanInterface.tsx [hunk 3]
```diff
@@ -51,34 +51,31 @@ export function OceanInterface (): JSX.Element | null {
   const transaction = useSelector((state: RootState) => firstTransactionSelector(state.ocean))
```
Comment (ivan-zynesis) [inline app/components/OceanInterface/OceanInterface.tsx:51]: Transaction (CTransactionSegwit) in read from queue is ready for broadcast

File: app/screens/AppNavigator/screens/Balances/ConvertScreen.tsx [hunk 4]
```diff
@@ -62,7 +62,6 @@ export function ConvertScreen (props: Props): JSX.Element {
       dispatch
     ).catch(e => {
       Logging.error(e)
-      dispatch(ocean.actions.setError(e))
```
Comment (ivan-zynesis) [inline app/screens/AppNavigator/screens/Balances/ConvertScreen.tsx:65]: signing error handled within prompt UI

File: app/translations/index.ts [hunk 5]
```diff
@@ -48,14 +48,16 @@ export function initI18n (): void {
...
-  const translation = i18n.translate(`${path}.${text}`)
-  if (translation !== null && translation !== undefined && translation !== '') {
-    return translation
-  } else {
-    return text
+  let translation = i18n.translate(`${path}.${text}`, options) ?? text
+  if (options !== undefined) {
+    // TODO(@ivan-zynesis): fix with i18n method
+    Object.keys(options).forEach(k => {
+      translation = translation.replaceAll(`%{${k}}`, options[k])
```
Comment (ivan-zynesis) [inline app/translations/index.ts:59]: @thedoublejay any suggestion?

File: app/api/scrypt.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      this.params.r,
+      this.params.p,
+      keyLength
+    )
+  }
+}
+
+export const scrypt = Platform.OS === 'web'
+  ? new Scrypt(new SimpleScryptsy())
+  : nativeScrypt
```
Comment (ivan-zynesis) [inline app/api/scrypt.ts:47]: https://forums.expo.dev/t/platform-specific-extensions/29337/3

Platform specific extension + ts + expo does not seems to work / supported yet

File: app/api/scrypt.native.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { ScryptProvider, Scrypt } from '@defichain/jellyfish-wallet-encrypted'
+import * as crypto from 'expo-crypto'
+
+// TODO: this is quick mock, required secured secret derivation
+class Mock implements ScryptProvider {
```
Comment (ivan-zynesis) [inline app/api/scrypt.native.ts:5]: As how it is named, this is not finalized. Major work/research require to make crypto safe randombytes and pbkdf2 works in mobile platforms.

NOTE: browserify randombytes do not work for mobile.

**Comment by fuxingloh** [comment]
Moved tailwind changes to https://github.com/DeFiCh/wallet/pull/372

**Comment by fuxingloh** [comment]
Moved translations changes to https://github.com/DeFiCh/wallet/pull/373

**Comment by fuxingloh** [comment]
Moved duplicated `WalletParamList` to https://github.com/DeFiCh/wallet/pull/374

**Comment by fuxingloh** [comment]
Moved `WalletProvider` refactor into https://github.com/DeFiCh/wallet/pull/380 without implementing the interfaces for it.

@ivan-zynesis 
- [ ] You should separate out the ocean interfaces changes into another PR and get that merged ASP
- [ ] Implement encrypted wallet with passphrase prompt. Look at `WalletContext.tsx` under encrypted wallet section. Added a simple implementation suggestion with the use of callback.

**Comment by ivan-zynesis** [comment]
Replaced with #403

**Pull request closed without merging**